src/main/java/com/google/maps/GeoApiContext.java

```diff
@@ -264,140 +273,226 @@ private void checkContext(boolean canUseClientId) {
-   * maxQps}).
-   */
-  public GeoApiContext setQueryRateLimit(int maxQps) {
-    requestHandler.setQueriesPerSecond(maxQps);
-    return this;
-  }
+    /**
+     * Sets the default write timeout for new connections. A value of 0 means no timeout.
+     *
+     * @param timeout The write timeout period in {@code unit}s.
+     * @param unit The write timeout time unit.
+     * @return Returns this builder for call chaining.
+     */
+    public Builder writeTimeout(long timeout, TimeUnit unit) {
+      builder.writeTimeout(timeout, unit);
+      return this;
+    }
 
-  /**
-   * Sets the rate at which queries are executed.
-   *
-   * @param maxQps The maximum number of queries to execute per second.
-   * @param minimumInterval The minimum amount of time, in milliseconds, to pause between requests.
-   *     Note that this pause only occurs if the amount of time between requests has not elapsed
-   *     naturally.
-   * @deprecated Please use {@link #setQueryRateLimit(int)} instead.
-   */
-  @Deprecated
-  public GeoApiContext setQueryRateLimit(int maxQps, int minimumInterval) {
-    requestHandler.setQueriesPerSecond(maxQps, minimumInterval);
-    return this;
-  }
+    /**
+     * Sets the cumulative time limit for which retry-able errors will be retried. Defaults to 60
+     * seconds. Set to zero to retry requests forever.
+     *
+     * <p>This operates separately from the count-based {@link #maxRetries(Integer)}.
+     *
+     * @param timeout The retry timeout period in {@code unit}s.
+     * @param unit The retry timeout time unit.
+     * @return Returns this builder for call chaining.
+     */
+    public Builder retryTimeout(long timeout, TimeUnit unit) {
+      this.errorTimeout = unit.toMillis(timeout);
+      return this;
+    }
 
-  /** Allows specific API exceptions to be retried or not retried. */
-  public GeoApiContext toggleifExceptionIsAllowedToRetry(
-      Class<? extends ApiException> exception, boolean allowedToRetry) {
-    if (allowedToRetry) {
-      exceptionsAllowedToRetry.add(exception);
-    } else {
-      exceptionsAllowedToRetry.remove(exception);
+    /**
+     * Sets the maximum number of times each retry-able errors will be retried. Set this to null to
+     * not have a max number. Set this to zero to disable retries.
+     *
+     * <p>This operates separately from the time-based {@link #retryTimeout(long, TimeUnit)}.
+     *
+     * @param maxRetries The maximum number of times to retry.
+     * @return Returns this builder for call chaining.
+     */
+    public Builder maxRetries(Integer maxRetries) {
+      this.maxRetries = maxRetries;
+      return this;
     }
-    return this;
-  }
 
-  /**
-   * Sets the proxy for new connections.
-   *
-   * @param proxy The proxy to be used by the underlying HTTP client.
-   */
-  public GeoApiContext setProxy(Proxy proxy) {
-    requestHandler.setProxy(proxy == null ? Proxy.NO_PROXY : proxy);
-    return this;
+    /**
+     * Disable retries completely.
```

Comment:
you might note that this clears the time limit and retries.

My 2c: I find this method a bit hard to grok. Setting retries to zero should be just as effective (even with e.g., a ten hour timeout)—can we just get users to do that?